deployment/hasura/migrations/AerieMerlin/16_associate_external_datasets_with_simulation/up.sql

```diff
@@ -0,0 +1,16 @@
+alter table plan_dataset
+  add column associated_simulation_dataset_id integer default null,
+  add constraint simulation_dataset_references_simulation_dataset
+    foreign key (associated_simulation_dataset_id)
+    references simulation_dataset
+    on update cascade
+    on delete cascade;
```

Comment:
I don't think the original use-case intended to change association, but I think supporting it is a good idea. However, wouldn't using `on delete set null` inadvertently associate any related external datasets with *all* simulation datasets whenever a sim dataset is deleted?

Reply:
It would. The question I should've asked is: "Do we always want to delete a plan dataset when its currently associated sim dataset is deleted?"

Reply:
Maybe we can just remove the `on delete` clause then. Since the end-user is responsible for creating, uploading, and associating these datasets, maybe they can also be responsible for updating the `simulation_dataset_id` column with a simple mutation instead of that behavior being hardcoded within the merlin db?

Reply:
That will cause exceptions due to constraint violations.

From the[ Postgres Docs](https://www.postgresql.org/docs/14/ddl-constraints.html):
> Restricting and cascading deletes are the two most common options. RESTRICT prevents deletion of a referenced row. NO ACTION means that if any referencing rows still exist when the constraint is checked, an error is raised; this is the default behavior if you do not specify anything. (The essential difference between these two choices is that NO ACTION allows the check to be deferred until later in the transaction, whereas RESTRICT does not.) CASCADE specifies that when a referenced row is deleted, row(s) referencing it should be automatically deleted as well. There are two other options: SET NULL and SET DEFAULT. These cause the referencing column(s) in the referencing row(s) to be set to nulls or their default values, respectively, when the referenced row is deleted. Note that these do not excuse you from observing any constraints. For example, if an action specifies SET DEFAULT but the default value would not satisfy the foreign key constraint, the operation will fail.

Reply:
I think `on delete cascade` makes the most sense then. `set null` would result in the external dataset being wrongly associated with other sim datasets unless explicitly updated, which I think is less intuitive than deletion.